src/thumbnail-generator.js

```diff
@@ -524,6 +524,7 @@ ThumbnailGenerator.prototype._BadStatusCodeException = function(statusCode) {
 util.inherits(ThumbnailGenerator.prototype._BadStatusCodeException, Error);
 
 ThumbnailGenerator.prototype._getExtension = function(name) {
+	name = name.replace(/\?.*$/, '')
```

Comment:
Thanks. I think actually we can make this a bit more efficient with a regex.

Please could you add

```js
const extensionRegex = /\.([^.?#;]+)[^.]*$/;
```

below `ffmpegTimeout` at the top

and then here we can just do
```js
const res = extensionRegex.exec(name);
return res ? res[1] : '';
```

Reply:
done I think this should be what you want